### PR TITLE
[FLINK-9446][docs] Update savepoint compatibility table for 1.4

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR updates the savepoint compatibility table for 1.4.

In FLINK-7461 we removed compatibility for 1.1.
In FLINK-7511 we removed CEP compatibility for 1.2.
In FLINK-8451 a compatibility issue for scala cases classes was identified.

/cc @StefanRRichter @dawidwys @aljoscha @twalthr

### CODE REVIEW

File: docs/ops/upgrading.md [hunk 1]
```diff
@@ -227,15 +229,25 @@ Savepoints are compatible across Flink versions as indicated by the table below:
...
+          time is not supported. Users have to first take a savepoint after migrating to Flink 1.3.x+, and then change
+          parallelism. Savepoints taken by the CEP library cannot be restored in 1.4.x+.</td>
     </tr>
     <tr>
           <td class="text-center"><strong>1.3.x</strong></td>
           <td class="text-center"></td>
           <td class="text-center"></td>
           <td class="text-center">O</td>
+          <td class="text-center">O</td>
+          <td class="text-left">Migrating from Flink 1.3.0 to Flink 1.4.[0,1] will fail if the savepoint contains scala case classes. Users have to either first upgrade to 1.3.1, or directlry to 1.4.2+.</td>
```
Comment (StefanRRichter) [inline docs/ops/upgrading.md]: typo `directlry `

File: docs/ops/upgrading.md [hunk 2]
```diff
@@ -227,15 +229,25 @@ Savepoints are compatible across Flink versions as indicated by the table below:
           <td class="text-center"></td>
           <td class="text-center">O</td>
           <td class="text-center">O</td>
-          <td class="text-left">When migrating from Flink 1.2.x to Flink 1.3.x, changing parallelism at the same
-          time is not supported. Users have to first take a savepoint after migrating to Flink 1.3.x, and then change
-          parallelism.</td>
+          <td class="text-center">O</td>
+          <td class="text-left">When migrating from Flink 1.2.x to Flink 1.3.x+, changing parallelism at the same
+          time is not supported. Users have to first take a savepoint after migrating to Flink 1.3.x+, and then change
+          parallelism. Savepoints taken by the CEP library cannot be restored in 1.4.x+.</td>
```
Comment (StefanRRichter) [inline docs/ops/upgrading.md]: "taken by" sounds a bit strange, because it is not the CEP library that takes them.

**Comment by StefanRRichter** [comment]
Update looks good from my side.

**Comment by dawidwys** [comment]
+1 from side

File: docs/ops/upgrading.md (hunk 2 above)
Comment (zentol) [reply docs/ops/upgrading.md]: How about `Savepoints taken for CEP applications ...`?

Comment (dawidwys) [reply docs/ops/upgrading.md]: Another suggestion : `Savepoints taken for applications using CEP library.`

File: docs/ops/upgrading.md [hunk 3]
```diff
@@ -227,15 +229,25 @@ Savepoints are compatible across Flink versions as indicated by the table below:
...
     </tr>
     <tr>
           <td class="text-center"><strong>1.3.x</strong></td>
           <td class="text-center"></td>
           <td class="text-center"></td>
           <td class="text-center">O</td>
+          <td class="text-center">O</td>
+          <td class="text-left">Migrating from Flink 1.3.0 to Flink 1.4.[0,1] will fail if the savepoint contains scala case classes. Users have to directly migrate to 1.4.2+ instead.</td>
```
Comment (twalthr) [inline docs/ops/upgrading.md]: Sounds good. nit: `Scala` instead of `scala`

**Pull request merged**